ts/features/wallet/component/FeaturedCardCarousel.tsx

```diff
@@ -52,9 +52,12 @@ const FeaturedCardCarousel: React.FunctionComponent<Props> = (props: Props) => {
     });
   }
 
-  const anyBonusNotActive =
-    (!pot.getOrElse(props.bpdActiveBonus, false) && bpdEnabled) ||
-    (!props.cgnActiveBonus && cgnEnabled);
+  const hasBpdActive: boolean | undefined = pot.getOrElse(
```

Comment:
this expression never returns `undefined`. It returns always a `boolean`

we should implement this logic

> if we have a value (pot some) we can say has bpd or not (true / false, depending of the pot value). Otherwise we can't say nothing (undefined)

```suggestion
  const hasBpdActive: boolean | undefined =
    props.bpdActiveBonus.kind === "PotSome"
      ? props.bpdActiveBonus.value
      : undefined;
```

Reply:
You're absolutely right I thought to commit the previously suggested code:
```typescript
const hasBpdActive: boolean | undefined =
    pot.isSome(props.bpdActiveBonus)
      ? props.bpdActiveBonus.value
      : undefined;
```

I'm fixing it asap!